security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationResources.scala

```diff
@@ -46,18 +59,81 @@ trait AuthenticationResources extends Directives with FailFastCirceSupport {
   protected lazy val additionalRoute: Route = Directives.reject
 }
 
+object AuthenticationResources {
+
+  def apply(config: Config, classLoader: ClassLoader, sttpBackend: SttpBackend[Future, Any])(
+      implicit ec: ExecutionContext
+  ): AuthenticationResources = {
+    implicit val sttpBackendImplicit: SttpBackend[Future, Any] = sttpBackend
+    AuthenticationProvider(config, classLoader)
+      .createAuthenticationResources(config, classLoader)
+  }
+
+}
+
 trait AnonymousAccess extends Directives {
-  val anonymousUserRole: Option[String]
+  this: AuthenticationResources =>
+
+  protected def anonymousUserRole: Option[String]
```

Comment:
Let's add a TODO, that it shouldn't be in this interface. We should have some external configuration for this instead of adding this field to each security extension's configuration.

Reply:
I've refactored it a little bit